### PR TITLE
Update py script to source Classes from Base/Sub files, content filtering, output controls and more

### PR DESCRIPTION
**Pull request by stevenjonescgm**
I added to the `create_compendiums.py` script because I wanted a compendium that excluded the Modern content and (optionally) flattened out subclasses.

I'm happy to make changes before merging upstream, or I'll just continue to maintain my fork for personal use.

Here are the changes summarized:

1. Content: Fixed file names in Homebrew and Unearthed Arcana
2. Script: compiles an `Unearthed Arcana Compendium.xml` but not add that to `Full Compendium.xml` to be consistent with previous behavior
3. Content: Updated Templates.xml to include notes for values and reference newer app features
4. Script: load and process partial files for BaseClasses and SubClasses
    1. using `<baseclass><name>Fighter</name>...` `<subclass baseclass="Fighter">...` as a replacement for `<class>...`
    2. Base/Sub files expected in directory `Character/Classes/<class-name>/<class-name> (<subclass-name>).xml`
5. Script: added argument processing and standard help features `python ./create_compendiums.py -h`
6 Script: add flags to manage whether subclasses are combined into one and/or are compiled separately (with or without base included) `--basetype-format=complete --subtype-format=reference` etc
7. Script: add flags so that certain tagged content can be excluded from compendiums
    1. the script looks for XML attributes which I've named `['UA', 'MF', 'HB', 'PS', 'IL']` for
        * UA Unearthed Arcana
        * MF Modern and Future
        * HB HomeBrew and 3rd Party
        * PS Psuedo Spells - class features added to the Spells list to compensate for lack of support in app, eg Maneuvers and Eldritch Invocations
        * IL Inline Lists - repeated class feature options for later levels (eg Warlock Eldritch Invocations and Sorcerer Metamagic. I'd prefer to use back-references, but upstream excludes level-prerequisite features from early auto-level blocks
    2. without using flags, MF and HB are excluded as they are not core D&D content

TODO:
* support subclasses against multiple baseclasses (Ranger).
* Consider how to get Mystic to work well with subclass pattern
* Investigate a scripting solution to item affixes until FightClub adds that cms feature

### CODE REVIEW

File: create_compendiums.py [hunk 1]
```diff
@@ -73,24 +163,54 @@ def create_category_compendiums():
...
+    categories = ['Items', 'Character', 'Spells', 'Bestiary', 'Unearthed Arcana']
     output_paths = []
     for category in categories:
         filenames = glob('%s/*.xml' % category)
         output_path = COMPENDIUM.format(category=category)
-        output_paths.append(output_path)
+
+        """build UA compendium, but exclude from Full"""
+        if output_path != 'Unearthed Arcana':
+            output_paths.append(output_path)
```
Comment (stevenjonescgm) [inline create_compendiums.py]: bug here. it was supposed to test `category != 'Unearthed Arcana'` to keep that out of Full

**Comment by stevenjonescgm** [comment]
added fix for Unearthed Arcana in Full. regenerated Compendiums for convenience of testing/reviewing. I don't really have a preference between including `Fighter Compendium.xml` and `Warlock Compendium.xml` etc in the repository. Could omit those (and eventually `Classes Compendium.xml`) and/or make a change to aggregate Fighter/Warlock etc into `Classes Compendium.xml` before `Full Compendium.xml`

**Comment by arpakarta** [comment]
I pulled out the remaining classes to Base/Sub pattern except

- original Ranger as I was not inclined to  make ranger subclasses work with both base classes nor encourage source duplication in this Base/Sub pattern)
- Mystic, which has subclasses that are not independent

I also added 
- an `IL="IL"` attribute to make the script optionally skip repeated class features (eg the Warlock Eldritch Invocations 
- parameters to the script allowing the subclass entries to be added as `reference` to the Compendium (directly without merging in baseclass)
- `<spellList>` elements allowing UA spell lists to live tagged as such and optionally included into the officially supported `<spell>...<class>...` list

**Comment by arpakarta** [comment]
the conflict in Ua Feats auto-resolves from command-line git. I've merged this branch and [ceryliae:master](/ceryliae/DnDAppFiles/) into [stevenjonescgm:master](/stevenjonescgm/DnDAppFiles/)

**Comment by stevenjonescgm** [comment]
This was a fun exercise in python, but superseded by dndbeyond when that launched many years ago

**Pull request closed without merging**